lib/js/components/Layout/Layout.vue

```diff
@@ -0,0 +1,261 @@
+<template>
+	<div class="layout">
+		<div v-if="sideMenuVisible" class="layout__navigation">
+			<div class="layout__mainMenu">
+				<slot name="mainMenu" />
+			</div>
+			<div class="layout__sideNav">
+				<slot name="sideNav" />
+			</div>
+		</div>
+		<div
+			class="layout__contentColumn"
+			:class="{
+				'-fullWidth': isContentFullWidth,
+				'-max900': !isContentFullWidth,
+				'-noPadding': contentWithoutPadding,
+			}"
+		>
+			<slot />
+		</div>
+		<template
+			v-if="
+				$slots.rightColumn &&
+				rightColumnVisible &&
+				!isContentFullWidth &&
+				rightColumnMode === LAYOUT_RIGHT_COLUMN_MODE.COLUMN_VISIBLE
+			"
+		>
+			<div class="layout__overlay" />
+			<div
+				class="layout__rightColumn"
+				:class="{
+					'-medium': rightColumnSize === LAYOUT_RIGHT_COLUMN_SIZE.MEDIUM,
+					'-large': rightColumnSize === LAYOUT_RIGHT_COLUMN_SIZE.LARGE,
+				}"
+			>
+				<slot name="rightColumn" />
+			</div>
+		</template>
+		<div
+			v-if="
+				$slots.sideBar &&
+				sideBarVisible &&
+				rightColumnMode === LAYOUT_RIGHT_COLUMN_MODE.SIDEBAR_VISIBLE &&
+				!isContentFullWidth
+			"
+			class="layout__sideBar"
+		>
+			<slot name="sideBar" />
+		</div>
+	</div>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/media-queries';
+
+$overlay-background-color: rgba(0, 0, 0, 0.7);
+
+$main-menu-width: 80px;
+
+$left-column-width: 23vw;
+$left-column-min-width: 200px;
+$left-column-max-width: 400px;
+
+$content-column-max-width: 900px;
+$content-column-full-max-width: 1280px;
+
+$right-column-medium-max-width: 400px;
+$right-column-medium-l-width: 30vw;
+$right-column-medium-l-min-width: 320px;
+$right-column-medium-l-max-width: 400px;
+$right-column-large-max-width: 560px;
+$right-column-large-l-width: 30vw;
+$right-column-large-l-min-width: 320px;
+$right-column-large-l-max-width: 560px;
+
+$side-bar-width: 66px;
+
+.layout {
+	$root: &;
+
+	display: flex;
+	flex-wrap: nowrap;
+	height: auto;
+	justify-content: space-between;
+	padding: 0;
+	position: relative;
+	width: 100%;
+
+	&__overlay {
+		background: rgba(0, 0, 0, 0.7);
+		display: none;
+		height: 100%;
+		position: absolute;
+		width: 100%;
+
+		@media #{breakpoint-s()} {
+			display: block;
+		}
+
+		@media #{breakpoint-l()} {
+			display: none;
+		}
+	}
+
+	&__navigation {
+		display: none;
```

Comment:
it isn't possible to show `__navigation` now on mobile. If I remember correctly, we agreed to show it on an overlay